src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -43,35 +42,53 @@ def initialise(self):
 
     def step(self, t, t_new, current_rv, **kwargs):
         """Gaussian IVP filter step as nonlinear Kalman filtering with zero data."""
+        # 0. Obtain the diffusion matrix; required for calibration / error estimation
+        discrete_dynamics = self.gfilt.dynamod.discretise(t_new - t)
+        diffmat = discrete_dynamics.diffusionmatrix(t_new)
+
+        # 1. Predict
         pred_rv, _ = self.gfilt.predict(t, t_new, current_rv, **kwargs)
+
+        # 2. Measure
+        meas_rv, info = self.gfilt.measmod.transition_rv(pred_rv, t_new, **kwargs)
+
+        # 3. Estimate the diffusion (sigma squared)
+        self.sigma_squared_mle = self._estimate_diffusion(pred_rv, meas_rv)
+        # 3.1. Adjust the prediction covariance to include the diffusion
+        pred_rv = Normal(
+            pred_rv.mean, pred_rv.cov + (self.sigma_squared_mle - 1) * diffmat
+        )
+        # 3.2 Update the measurement covariance (measure again)
+        meas_rv, info = self.gfilt.measmod.transition_rv(pred_rv, t_new, **kwargs)
+
+        # 4. Update
         zero_data = 0.0
         filt_rv, meas_rv, info = self.gfilt.update(t_new, pred_rv, zero_data, **kwargs)
```

Comment:
is there a reason for calling a full update after the meas_rv is already computed (and updated!)? If I remember correctly, this update will call yet another `measmod.transition_rv`. It might be interesting to extract the update step from the update step (latter I think of as measurement + update, you know what I mean). This can then be called here.

Reply:
No real reason, I just didn't want to change too many things at the same time. But you're right, that should maybe still be included in this PR!

Should that change happen in `Kalman` or in `BayesFiltSmooth`? I would add a new method `measure(time, randvar)` and another one that performs only the conditioning, which are then both called in `update`. Do you have suggestions on the naming? Maybe something like `condition(state, measurement, crosscov)`? But maybe that sounds a bit too general

Reply:
I guess it is best placed in `Kalman`? What I would do is split the `Kalman.update` in `measure` and sth like `condition_prediction` or so, the latter of which contains the linear algebra things and can be called again here. Naming? I dont know, in my head "update" is "measure" and "update", but in code this is clearly not helpful :D Maybe something really verbose until we have a better idea, like `condition_prediction_on_measurement` or so?

Reply:
Thanks a lot for the suggestions! I split up the update into measure and condition_state_on_measurement, and I call those in the ODESolver.step now :)